File: fingerprint/src/main/java/com/yf/verify/fingerprint/factory/BiometricPromptFactory.java
package com.yf.verify.fingerprint.factory;

import android.os.Build;
import android.support.v4.app.FragmentActivity;
import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;

import com.yf.verify.callback.FingerprintAuthenticatedCallback;

import javax.crypto.Cipher;

public class BiometricPromptFactory {
    private FragmentActivity activity;
    private Cipher cipher;
    private FingerprintManagerCompat mFingerprintManager;
    private FingerprintAuthenticatedCallback fingerprintCallback;
    private BaseBiometricPrompt prompt;

    public BiometricPromptFactory(FragmentActivity activity, Cipher cipher, FingerprintManagerCompat mFingerprintManager, FingerprintAuthenticatedCallback fingerprintCallback) {
        this.activity = activity;
        this.cipher = cipher;
        this.mFingerprintManager = mFingerprintManager;
        this.fingerprintCallback = fingerprintCallback;
    }


    public void execute(int version) {
        if (version >= Build.VERSION_CODES.P) {
            prompt = new BaseBiometricPrompt29(activity, fingerprintCallback);
//            prompt = new BaseBiometricPrompt28(activity, cipher, mFingerprintManager, fingerprintCallback);
        } else if (version >= Build.VERSION_CODES.M) {
            prompt = new BaseBiometricPrompt28(activity, cipher, mFingerprintManager, fingerprintCallback);
        } else {
            if (null != fingerprintCallback) {
                fingerprintCallback.noEnrolledFingerprints();
            }
            return;
        }
        prompt.show();
    }
}

File: fingerprint/src/main/java/com/yf/verify/fingerprint/factory/BiometricPromptFactory.kt
package com.yf.verify.fingerprint.factory

import android.os.Build
import android.support.v4.app.FragmentActivity
import com.yf.verify.callback.FingerprintAuthenticatedCallback
import com.yf.verify.callback.UpdateFingerprintCallback
import com.yf.verify.fingerprint.FingerprintAuthenticationDialogFragment
import javax.crypto.Cipher

class BiometricPromptFactory(var activity: FragmentActivity, var cipher: Cipher, var stage: FingerprintAuthenticationDialogFragment.Stage, var updateFingerprintCallback: UpdateFingerprintCallback, var fingerprintCallback: FingerprintAuthenticatedCallback) {
    val prompt = hashMapOf<Int, BaseBiometricPrompt>()

    init {
        prompt[Build.VERSION_CODES.P] = BaseBiometricPrompt29(activity, fingerprintCallback)
        prompt[Build.VERSION_CODES.M] = BaseBiometricPrompt28(activity, cipher, stage, updateFingerprintCallback, fingerprintCallback)
    }

    fun execute(version: Int) {
        if (version >= Build.VERSION_CODES.P) {
            prompt[Build.VERSION_CODES.P]?.show()
        } else {
            prompt[Build.VERSION_CODES.M]?.show()
        }
    }
}